addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/generators/BodyGenerator.java

```diff
@@ -66,10 +67,41 @@ public BodyGenerator(Generators generators) {
                         }
                     });
 
-    public String generate(String name, boolean isArray, List<String> refs) {
+    public String generateBodyWithObjectMaps(String name, boolean isArray, List<String> refs) {
```

Comment:
I'd leave it as `generate` (same for new method), the parameters already imply what's being used to generate the body.